### ISSUE TITLE
And How it goes to wallet ?

### ISSUE THREAD

**Issue by amaendeepm** [codebasics/cool_python_apps#4]
How this code https://github.com/codebasics/cool_python_apps/blob/main/2_bitcoin_mining/bitcoin_mining.py sends mined qty to wallet ?

**Comment by zevaryx** [comment]
This isn't a true Bitcoin miner, but more of a proof of concept on how mining works. It only mines a single simulated block

**Comment by AdrianViverosL** [comment]
Yeah! as zevaryx mentioned above, this code only has demo purposes to get the idea and concept of mining.

**Comment by bener07** [comment]
And where can I find a TRUE script to mine bitcoins to my wallet, I changed the code, and now is in a loop so is just mining and mining and all of that. so where can I find one that sends the bitcoins to my wallet?

**Comment by zevaryx** [comment]
For various reasons, that would be a terrible idea.

tl;dr is that you can't directly mine BTC with standard computers anymore, ignoring that this code is just a proof of concept.

Full reasons:
* Bitcoin mining is astronomically difficult to compute with normal computers, even with GPU acceleration. You typically mine it with something called an [ASIC](https://en.wikipedia.org/wiki/Application-specific_integrated_circuit) miner, but not with a normal PC
* Most mining programs are written in much lower languages than Python, like C/C++, and have very well-optimized algorithms that they use, typically taking advantage of [SSE](https://en.wikipedia.org/wiki/Streaming_SIMD_Extensions) and [AVX](https://en.wikipedia.org/wiki/Advanced_Vector_Extensions) instructions on CPUs, and [OpenCL](https://en.wikipedia.org/wiki/OpenCL) or [CUDA](https://en.wikipedia.org/wiki/CUDA) processing on GPUs. This type of optimization is not possible with Python (at least, [not without writing a good portion of the code in C and importing it](https://docs.python.org/3/extending/extending.html))
* This code doesn't actually mine anything, but merely simulates what mining would be like. Mining requires accessing the [blockchain](https://en.wikipedia.org/wiki/Blockchain), synchronizing, validation, and mining. This code does none of that, merely simulating something similar to mining.
